File: arewepythonyet/bench/b_bridge/sumlog.js

function sum_log(iterations) {
  var total = 0
  for (var i = 1; i <= iterations; i++) {
    total += Math.log(i)
  }
  return total
}

for (var i = 0; i < 3; i++) {
  t1 = +Date.now()
  sum_log(1000000)
  t2 = +Date.now()
  print((t2 - t1) / 1000)
}

File: arewepythonyet/bench/b_bridge/sumlog.py

import js
import time

Math = js.globals["Math"]

def sum_log(iterations):
    total = 0
    i = 1
    while i <= iterations:
        total += float(Math.log(i))
        i += 1
    return total


for i in xrange(3):
    t1 = time.time()
    sum_log(1000000)
    t2 = time.time()
    print t2 - t1

